src/main/java/be/seeseemelk/mockbukkit/adventure/PlainTextComponentProviderImpl.java

```diff
@@ -0,0 +1,110 @@
+package be.seeseemelk.mockbukkit.adventure;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TranslatableComponent;
+import net.kyori.adventure.text.TranslationArgument;
+import net.kyori.adventure.text.flattener.ComponentFlattener;
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+import net.kyori.adventure.translation.GlobalTranslator;
+import net.kyori.adventure.translation.TranslationRegistry;
+import net.kyori.adventure.translation.Translator;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PlainTextComponentProviderImpl implements PlainTextComponentSerializer.Provider
+{
+
+	private static final Pattern LOCALIZATION_PATTERN = Pattern.compile("%(?:(\\d+)\\$)?s");
+
+	private static final ComponentFlattener BASIC_COMPONENT_FLATTENER = ComponentFlattener.basic().toBuilder()
+			.complexMapper(TranslatableComponent.class, PlainTextComponentProviderImpl::processTranslatable).build();
+
+	@Override
+	public @NotNull PlainTextComponentSerializer plainTextSimple()
+	{
+		return PlainTextComponentSerializer.builder()
+				.flattener(BASIC_COMPONENT_FLATTENER)
+				.build();
+	}
+
+	@Override
+	public @NotNull Consumer<PlainTextComponentSerializer.Builder> plainText()
+	{
+		return builder -> builder.flattener(BASIC_COMPONENT_FLATTENER);
+	}
+
+	private static void processTranslatable(TranslatableComponent translatable, Consumer<Component> consumer)
```

Comment:
Please split this method into multiple methods to reduce cognitive complexity

Reply:
Okay, moved some of the code into a method